src/Resolvers/Microsoft.DotNet.NativeWrapper/NETCoreSdkResolverNativeWrapper.cs

```diff
@@ -50,5 +50,32 @@ public static string[] GetAvailableSdks(string dotnetExeDirectory)
 
             return list.Entries;
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct hostfxr_initialize_parameters
+        {
+
+        }
+
+        public static int InitializeForRuntimeConfig(string runtimeConfigPath)
+        {
+            var result = -1;
+            IntPtr hostContextHandle = default;
+
+            hostfxr_initialize_parameters parameters = new hostfxr_initialize_parameters();
+
+            IntPtr parametersPtr = Marshal.AllocHGlobal(Marshal.SizeOf(parameters));
+            Marshal.StructureToPtr(parameters, parametersPtr, false);
+
+            if (File.Exists(runtimeConfigPath))
+            {
+                result = Interop.RunningOnWindows
+                    ? Interop.Windows.hostfxr_initialize_for_runtime_config(runtimeConfigPath, default, out hostContextHandle)
+                    : Interop.Unix.hostfxr_initialize_for_runtime_config(runtimeConfigPath, default, out hostContextHandle);
+            }
+
+            Marshal.FreeHGlobal(parametersPtr);
+            return result;
```

Comment:
The `hostContextHandle` is never closed. This at the very least means that the method will really only work once.
@elinor-fung could you check if the host will work correctly if this calls `hostfxr_initialize_for_runtime_config` followed by `hostfxr_close` multiple times over? I think it should - but we might not have a test for that case.